motor/frameworks/asyncio/__init__.py

```diff
@@ -37,7 +37,11 @@
 
 
 def get_event_loop():
-    return asyncio.get_event_loop()
+    try:
+        return asyncio.get_running_loop()
+    except RuntimeError:
+        # Workaround for bugs.python.org/issue39529.
+        return asyncio.get_event_loop_policy().get_event_loop()
```

Comment:
I was expecting `asyncio.set_event_loop(asyncio.new_event_loop())`. Doesn't `AbstractEventLoopPolicy.get_event_loop()` have the same behavior as asyncio.get_event_loop()?

Did the maintainers just forget to deprecate AbstractEventLoopPolicy.get_event_loop?

Reply:
Yes, it hasn't been deprecated yet.  This is the workaround that tornado is using for their 6.2 release.

Reply:
Oh I see, `asyncio.set_event_loop(asyncio.new_event_loop())` will incorrectly reinitialize the loop when there is a loop already set on the thread, but it just hasn't been started yet:
```
>>> asyncio.get_running_loop()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: no running event loop
>>> id(asyncio.get_event_loop())
4361557568
>>> asyncio.set_event_loop(asyncio.new_event_loop())
>>> id(asyncio.get_event_loop())
4374658448
```

Reply:
@blink1073 this workaround in tornado is only used for the deprecated IOLoop().start() and it is only intended to support

```
with warnings.catch_warnings():
    warnings.simplefilter("ignore")
    loop = IOLoop()
    
...
loop.start()
```

Reply:
anywhere the user creates an IOLoop still happens via `asyncio.get_event_loop()` so if the user creates an IOLoop too early they get a DeprecationWarning

https://github.com/tornadoweb/tornado/blob/713f06b115f7125b4cf0d21d667ab5c801318baf/tornado/platform/asyncio.py#L295-L297